libssh2/libssh2.cabal

```diff
@@ -64,7 +64,7 @@ Library
                        network >= 2.3 && < 3.2,
                        syb >= 0.3.3, time >= 1.2,
                        bytestring >= 0.9,
-                       unix
+                       unix, base64, utf8-string
```

Comment:
Not sure how you feel about these extra dependencies, but they allow us to keep our types mostly unchanged.